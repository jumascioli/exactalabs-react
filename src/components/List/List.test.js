import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import List from '.';

const tasks = [
  {
    id: "123456",
    title: "Teste tasks",
  },
];

describe("<List />", () => {

  it("should render correctly", () => {
    const { container } = render(<List tasks={tasks} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render item", () => {
    render(<List tasks={tasks} />);

    const item = screen.getByText(/Teste tasks/i);

    expect(item).toBeInTheDocument();
  });

  it("should call onDelete", () => {
    const handleDelete = jest.fn(); 

    render(<List tasks={tasks} onDelete={handleDelete} />);

    const button = screen.getByRole("button");

    userEvent.click(button);

    expect(handleDelete).toBeCalledWith("123456");
  });

  it("should render empty", () => {
    render(<List tasks={[]} />);

    const button = screen.queryByRole("button");

    expect(button).not.toBeInTheDocument();
  });
});