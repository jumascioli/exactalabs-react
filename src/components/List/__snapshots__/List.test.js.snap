// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> should render correctly 1`] = `
<ul
  class="sc-bdfBQB ivTgth"
>
  <li
    class="sc-gsTEea eaPawV"
  >
    Teste tasks
    <button
      class="sc-dlfnuX hhtfFF"
    />
  </li>
</ul>
`;
